src/main/kotlin/sliv/tool/scene/view/virtualizedfx/VirtualizedFXGrid.kt

```diff
@@ -0,0 +1,47 @@
+package sliv.tool.scene.view.virtualizedfx
+
+import io.github.palexdev.mfxcore.base.beans.Position
+import io.github.palexdev.virtualizedfx.controls.VirtualScrollPane
+import io.github.palexdev.virtualizedfx.grid.VirtualGrid
+import javafx.event.EventHandler
+import javafx.geometry.Orientation
+import javafx.scene.Node
+import javafx.scene.input.ScrollEvent
+import sliv.tool.scene.model.VisualizationFrame
+import sliv.tool.scene.view.Grid
+
+class VirtualizedFXGrid(
+    private val virtualGrid: VirtualGrid<VisualizationFrame, FrameViewAdapter>, private val vsp: VirtualScrollPane
+) : Grid {
+    private var dragStart = Position.of(-1.0, -1.0)
+    private var initialValues = Position.of(0.0, 0.0)
+
+    override fun setUpPanning() {
+        virtualGrid.setOnMousePressed { event ->
+            dragStart = Position.of(event.x, event.y)
+            initialValues = virtualGrid.position
+        }
+
+        virtualGrid.setOnMouseDragged { event ->
+            val xDelta = -(event.x - dragStart.x)
+            val yDelta = -(event.y - dragStart.y)
+            virtualGrid.scrollTo(initialValues.x + xDelta, Orientation.HORIZONTAL)
+            virtualGrid.scrollTo(initialValues.y + yDelta, Orientation.VERTICAL)
+        }
+    }
+
+    override fun setOnScroll(handler: EventHandler<ScrollEvent>) {
+        virtualGrid.addEventHandler(ScrollEvent.SCROLL, handler)
```

Comment:
what about side scrollers that can be used without mouse wheel? are they hidden?

Reply:
I consume scroll event by VirtualGrid to prevent scrolling when mouse is on frames. When mouse is on scrollbars they appears and handles scroll by itself, so mouse wheel provokes scrolling.